protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -105,6 +115,93 @@ struct ScarbStarknetContractArtifact {
     casm: Option<PathBuf>,
 }
 
+fn execute_syscall(
+    system: &ResOperand,
+    vm: &mut VirtualMachine,
+    blockifier_state: &mut CachedState<DictStateReader>,
+) -> Result<(), HintError> {
+    let (cell, offset) = extract_buffer(system);
+    let mut system_ptr = get_ptr(vm, cell, &offset)?;
+
+    let selector = felt_from_pointer(vm, &mut system_ptr)
+        .unwrap()
+        .to_bytes_be();
+
+    let gas_counter = usize_from_pointer(vm, &mut system_ptr).unwrap();
+    let contract_address = felt_from_pointer(vm, &mut system_ptr).unwrap();
+    let entry_point_selector = felt_from_pointer(vm, &mut system_ptr).unwrap();
+
+    let start = relocatable_from_pointer(vm, &mut system_ptr).unwrap();
+    let end = relocatable_from_pointer(vm, &mut system_ptr).unwrap();
+    let calldata = read_data_from_range(vm, start, end).unwrap();
+
+    assert_eq!(std::str::from_utf8(&selector).unwrap(), "CallContract");
+    let result = call_contract(
+        &contract_address,
+        &entry_point_selector,
+        &calldata,
+        blockifier_state,
+    )
+    .unwrap();
+
+    insert_at_pointer(vm, &mut system_ptr, gas_counter).unwrap();
+    insert_at_pointer(vm, &mut system_ptr, Felt252::from(0)).unwrap();
+
+    let mut ptr = vm.add_memory_segment();
+    let start = ptr;
+    for value in result {
+        insert_at_pointer(vm, &mut ptr, value).unwrap();
+    }
+    let end = ptr;
+
+    insert_at_pointer(vm, &mut system_ptr, start).unwrap();
+    insert_at_pointer(vm, &mut system_ptr, end).unwrap();
+
+    Ok(())
+}
+
+// This can mutate state, the name of the syscall is not very good
+fn call_contract(
+    contract_address: &Felt252,
+    entry_point_selector: &Felt252,
+    calldata: &[Felt252],
+    blockifier_state: &mut CachedState<DictStateReader>,
+) -> Result<Vec<Felt252>> {
+    let contract_address = ContractAddress(PatriciaKey::try_from(StarkFelt::new(
+        contract_address.to_be_bytes(),
+    )?)?);
+    let entry_point_selector =
+        EntryPointSelector(StarkHash::new(entry_point_selector.to_be_bytes())?);
+    let account_address = ContractAddress(patricia_key!(TEST_ACCOUNT_CONTRACT_ADDRESS));
+    let calldata = Calldata(Arc::new(
+        calldata
+            .iter()
+            .map(|data| StarkFelt::new(data.to_be_bytes()))
+            .collect::<Result<Vec<_>, _>>()?,
+    ));
+    let entry_point = CallEntryPoint {
+        class_hash: None,
+        code_address: Some(contract_address),
+        entry_point_type: EntryPointType::External,
+        entry_point_selector,
+        calldata,
+        storage_address: contract_address,
+        caller_address: account_address,
+        call_type: CallType::Call,
+    };
```

Comment:
Not sure if `CallEntryPoint` is a correct interface for that, but it seems to be working. 

We can create an issue to investigate that and follow up in the future if there are problems with tests.